includes/class-wc-post-data.php

```diff
@@ -121,13 +121,24 @@ public static function delete_product_query_transients() {
 	/**
 	 * Handle type changes.
 	 *
-	 * @since 3.0.0
+	 * @since 4.9.0
+	 *
 	 * @param WC_Product $product Product data.
 	 * @param string     $from    Origin type.
 	 * @param string     $to      New type.
 	 */
 	public static function product_type_changed( $product, $from, $to ) {
-		if ( 'variable' === $from && 'variable' !== $to ) {
+		/**
+		 * Filter to prevent variations from being deleted while switching from a variable product type to a variable product type.
+		 *
+		 * @since 4.9.0
+		 *
+		 * @param bool       A boolean value of true will delete the variations.
+		 * @param WC_Product $product Product data.
+		 * @return string    $from    Origin type.
```

Comment:
"Origin" or "Original"?

Reply:
"Origin" is the intentional use. It's directly copied from the "Origin type" text [above](https://github.com/woocommerce/woocommerce/blame/34563c22bb302a969da0d966884c4d17b020a950/includes/class-wc-post-data.php#L127) (line 127). IMO "origin" and "new" makes sense in the context of "from" and "to. What do you think?

Reply:
Ah didn't notice the parameter comment in the method doc block. Keeping it consistent makes sense so `Origin type` is fine.